Realm/RLMCollection.mm

```diff
@@ -149,6 +150,17 @@ - (NSUInteger)countByEnumeratingWithState:(NSFastEnumerationState *)state
         return results;
     }
 
+    for (RLMProperty *prop in info->rlmObjectSchema.swiftGenericProperties) {
+        if ([prop.name isEqual:key] && prop.swiftIvar && !prop.optional) {
```

Comment:
Why the optional check? `value(forKey:)` should work with optional properties, and you handle this case a little later with the `?: NSNull.null`.

Also shouldn't need to check for `prop.swiftIvar` since `swiftGenericProperties` checks for the swift ivars already.

In other words, this implementation should be equivalent:

```objc
for (RLMProperty *prop in info->rlmObjectSchema.swiftGenericProperties) {
    if (![prop.name isEqual:key])
        continue;

    for (size_t i = 0; i < count; i++) {
        size_t rowIndex = [collection indexInSource:i];
        RLMObjectBase *accessor = RLMCreateObjectAccessor(realm, *info, rowIndex);
        [results addObject:[accessor valueForKey:key] ?: NSNull.null];
    }
    return results;
}
```